### PR TITLE
ISSUE 3 - If someone says No to original message, not the reply from …

### PR DESCRIPTION
**Pull request by william-sheldon**
…the bot, it should delete the tweet

https://github.com/HUU/MegadomeSays/issues/3

### CODE REVIEW

File: lib.js [hunk 1]
```diff
@@ -18,11 +18,18 @@ var CreateTweetRecord = exports.CreateTweetRecord = async function(records, mess
     });
 }
 
-var ReactionShouldBeTweeted = exports.ReactionShouldBeTweeted = async function(records, reaction) {
+var NoTweetFound = exports.NoTweetFound = async function(records, reaction) {
     var record = await FetchTweetRecord(records, reaction.message.id);
     return !record.exists;
 }
 
+var GetKnownTweetUrl = exports.GetKnownTweetUrl = async function(records, reaction) {
```
Comment (william-sheldon) [inline lib.js:26]: Should be "GetPreviouslyTweetedTweetId"

File: bot.js [hunk 2]
```diff
@@ -33,11 +33,21 @@ discordClient.on('messageReactionAdd', async (reaction) => {
...
-			reaction.message.lineReplyNoMention("I deleted that tweet; please don't get Jimmy cancelled.")
+			await handleOneReactionAtATime.runExclusive(async () => {
+				if (await lib.NoTweetFound(records, reaction)) {
+					console.log(`Attempting to delete the tweet referenced in "${reaction.message.content}".`)
+					const tweetId = lib.ExtractTweetId(reaction.message.content)
+					await lib.DeleteTweet(twitterClient, tweetId);
+					console.log(`Tweet ${tweetId} deleted.`);
+					reaction.message.lineReplyNoMention("I deleted that tweet; please don't get Jimmy cancelled.")
+				} else {
+					var tweetId = await lib.GetKnownTweetUrl(records, reaction);
```
Comment (william-sheldon) [inline bot.js:44]: bro is it a constant or a var we will never know

File: bot.js [hunk 3]
```diff
@@ -33,11 +33,21 @@ discordClient.on('messageReactionAdd', async (reaction) => {
...
+					console.log(`Attempting to delete the tweet referenced in "${reaction.message.content}".`)
```
Comment (HUU) [inline bot.js:45]: Lots of code duplicated, everything is the same after you fetch the `tweetId`.

```
if (await lib.NoTweetFound(records, reaction)) {
    const tweetId = lib.ExtractTweetId(reaction.message.content);
} else {
   const tweetId = await lib.GetKnownTweetUrl(records, reaction);
}

// everything else..
```

File: lib.js [hunk 4]
```diff
@@ -18,11 +18,18 @@ var CreateTweetRecord = exports.CreateTweetRecord = async function(records, mess
...
+    var record = await FetchTweetRecord(records, reaction.message.id);
+    tweetUrl = record.get('Tweet');
```
[the remark this answers is not in the document]
Comment (william-sheldon) [reply lib.js:28]: is this as compressed as possible?

```suggestion
var GetKnownTweetUrl = exports.GetKnownTweetUrl = async function(records, reaction) {
    var record = await FetchTweetRecord(records, reaction.message.id);
    return ExtractTweetId(record.get('Tweet'));
}
```

File: bot.js (hunk 3 above)
Comment (william-sheldon) [reply bot.js:45]: Here's the new deleteThis

```suggestion
		} else if (reaction.emoji.name == 'deleteThis') {
			await handleOneReactionAtATime.runExclusive(async () => {
				if (await lib.NoTweetFound(records, reaction)) {
					var tweetId = lib.ExtractTweetId(reaction.message.content)
				} else {
					var tweetId = await lib.GetKnownTweetUrl(records, reaction);
				}
				await lib.DeleteTweet(twitterClient, tweetId);
				console.log(`Tweet ${tweetId} deleted.`);
				reaction.message.lineReplyNoMention("I deleted that tweet; please don't get Jimmy cancelled.")
			})
		}
	}
```

File: lib.js [hunk 5]
```diff
@@ -18,11 +18,18 @@ var CreateTweetRecord = exports.CreateTweetRecord = async function(records, mess
...
 }
 
-var ReactionShouldBeTweeted = exports.ReactionShouldBeTweeted = async function(records, reaction) {
+var NoTweetFound = exports.NoTweetFound = async function(records, reaction) {
     var record = await FetchTweetRecord(records, reaction.message.id);
     return !record.exists;
 }
 
+var GetKnownTweetUrl = exports.GetKnownTweetUrl = async function(records, reaction) {
+    var record = await FetchTweetRecord(records, reaction.message.id);
```
[the remark this answers is not in the document]
Comment (HUU) [reply lib.js:27]:
```suggestion
    const record = await FetchTweetRecord(records, reaction.message.id);
```

**Review by HUU — APPROVED** [review]

**Pull request merged**